src/App.tsx

```diff
@@ -1,155 +1,82 @@
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, { useEffect, useMemo, useState } from 'react';
+import { Todo } from './types/Todo';
+import { Error, Filter } from './utils/Enum';
+import { getTodos } from './api/todos';
+import { TodoFooter } from './components/TodoFooter';
+import { TodoHeder } from './components/TodoHeder';
+import { TodoList } from './components/TodoList';
+import { TodoError } from './components/TodoError';
 
-const USER_ID = 0;
+const USER_ID = 11211;
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [hasError, setHasError] = useState(Error.NOTHING);
+  const [filterTodo, setFilterTodo] = useState(Filter.ALL);
+  const [isActive, setIsActive] = useState(false);
+  const [isComplited, setIsComplited] = useState(false);
+
+  const filteredTodos = useMemo(() => {
+    switch (filterTodo) {
+      case Filter.COMPLETED:
+        return todos.filter(todo => todo.completed);
+
+      case Filter.ACTIVE:
+        return todos.filter(todo => !todo.completed);
+
+      default:
+        return todos;
+    }
+  }, [filterTodo, todos]);
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then((data) => setTodos(data))
+      .catch(() => setHasError(Error.FETCH));
+  }, []);
+
+  useEffect(() => {
+    if (!todos.find(todo => !todo.completed) && todos.length) {
+      setIsActive(true);
+    } else {
+      setIsActive(false);
+    }
```

Comment:
```suggestion
    setIsActive(!todos.find(todo => !todo.completed) && todos.length);
```

Reply:
Some problems with it:
Argument of type 'number | false' is not assignable to parameter of type 'SetStateAction<boolean>'.
Type 'number' is not assignable to type 'SetStateAction<boolean>'.
<img width="1010" alt="image" src="https://github.com/mate-academy/react_todo-app-loading-todos/assets/77534644/e075bb7b-93fc-4e4b-84bb-c12a99028bea">

Reply:
```
setIsActive(Boolean(!todos.find(todo => !todo.completed) && todos.length));
```
or
```
setIsActive(!!(!todos.find(todo => !todo.completed) && todos.length));
```